__tests__/utils/wallet.test.js

```diff
@@ -174,43 +176,175 @@ test('hd derivation', () => {
   // For 31 bytes private keys
-  const xpriv31B = wallet.getXPrivKeyFromSeed(Words31B);
+  const xpriv31B = wallet.getXPrivKeyFromSeed(Words31B, {networkName: 'testnet'});
   expect(xpriv31B.privateKey.bn.toBuffer().length).toBe(31);
   const dpriv31Bd = xpriv31B.derive('m/44\'/280\'/0\'/0');
   const dpriv31B = xpriv31B.deriveNonCompliantChild('m/44\'/280\'/0\'/0');
   for (let i = 0; i < 10; i++) {
-    const addrd = dpriv31Bd.derive(i).publicKey.toAddress().toString();
-    const addr = dpriv31B.deriveNonCompliantChild(i).publicKey.toAddress().toString();
+    const addrd = dpriv31Bd.derive(i).publicKey.toAddress(testnet.bitcoreNetwork).toString();
+    const addr = dpriv31B.deriveNonCompliantChild(i).publicKey.toAddress(testnet.bitcoreNetwork).toString();
     expect(addr).toStrictEqual(addrd); // derive and deriveNonCompliant should be the same
     expect(addr).toStrictEqual(Addr31B[i]); // and both should generate the expected address
   }
 
   // For 32 bytes private keys
-  const xpriv32B = wallet.getXPrivKeyFromSeed(Words32B);
+  const xpriv32B = wallet.getXPrivKeyFromSeed(Words32B, {networkName: 'testnet'});
   expect(xpriv32B.privateKey.bn.toBuffer().length).toBe(32);
   const dpriv32Bd = xpriv32B.derive('m/44\'/280\'/0\'/0');
   const dpriv32B = xpriv32B.deriveNonCompliantChild('m/44\'/280\'/0\'/0');
   for (let i = 0; i < 10; i++) {
-    const addrd = dpriv32Bd.derive(i).publicKey.toAddress().toString();
-    const addr = dpriv32B.deriveNonCompliantChild(i).publicKey.toAddress().toString();
+    const addrd = dpriv32Bd.derive(i).publicKey.toAddress(testnet.bitcoreNetwork).toString();
+    const addr = dpriv32B.deriveNonCompliantChild(i).publicKey.toAddress(testnet.bitcoreNetwork).toString();
     expect(addr).toStrictEqual(addrd);
     expect(addr).toStrictEqual(Addr32B[i]);
   }
 
   // test our address generation
   // XXX: the accountDerivationIndex being 0'/0 is a temporary fix
   // getXPubKeyFromSeed does not derive for 'change' (https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#change)
-  const xpub31B = wallet.getXPubKeyFromSeed(Words31B, {accountDerivationIndex: '0\'/0'});
+  const xpub31B = wallet.getXPubKeyFromSeed(Words31B, {networkName: 'testnet', accountDerivationIndex: '0\'/0'});
   const addressObj31B = wallet.getAddresses(xpub31B, 0, 10, 'testnet');
   const address31B = Object.keys(addressObj31B);
   for (let i = 0; i < 10; i++) {
     expect(address31B[i]).toStrictEqual(Addr31B[i]);
   }
 
-  const xpub32B = wallet.getXPubKeyFromSeed(Words32B, {accountDerivationIndex: '0\'/0'});
+  const xpub32B = wallet.getXPubKeyFromSeed(Words32B, {networkName: 'testnet', accountDerivationIndex: '0\'/0'});
   const addressObj32B = wallet.getAddresses(xpub32B, 0, 10, 'testnet');
   const address32B = Object.keys(addressObj32B);
   for (let i = 0; i < 10; i++) {
     expect(address32B[i]).toStrictEqual(Addr32B[i]);
   }
 });
+
+test('testnet: hd derivation with password', () => {
```

Comment:
Change to passphrase. I know that it is confusing but, in our wallets, password, PIN, and passphrase have very different meanings.